homeassistant/components/bluesound/media_player.py

```diff
@@ -312,20 +311,24 @@ async def force_update_sync_status(self) -> bool:
 
     async def _start_poll_command(self):
         """Loop which polls the status of the player."""
-        try:
-            while True:
+        while True:
+            try:
                 await self.async_update_status()
-
-        except (TimeoutError, ClientError):
-            _LOGGER.error("Node %s:%s is offline, retrying later", self.host, self.port)
-            await asyncio.sleep(NODE_OFFLINE_CHECK_TIMEOUT)
-            self.start_polling()
-
-        except CancelledError:
-            _LOGGER.debug("Stopping the polling of node %s:%s", self.host, self.port)
-        except Exception:
-            _LOGGER.exception("Unexpected error in %s:%s", self.host, self.port)
-            raise
+            except (TimeoutError, ClientError):
+                _LOGGER.error(
+                    "Node %s:%s is offline, retrying later", self.host, self.port
+                )
+                await asyncio.sleep(NODE_OFFLINE_CHECK_TIMEOUT)
+            except CancelledError:
+                _LOGGER.debug(
+                    "Stopping the polling of node %s:%s", self.host, self.port
+                )
+                return
+            except Exception:
```

Comment:
In the future, can the library raise more specific errors?

Reply:
Maybe something like `received unexpected response from player` when the response parsing fails, but nothing we can act on in home assistant. Or something like a `PlayerError`, which gets raised whenever anything during the request fails.

Reply:
I mean more like, currently we catch 4 exceptions, it would be nice if the library could for example catch `TimeoutError` and `ClientError` and raise `NodeOfflineError`. This example is a nice to have, but it would be nice to give that bit of context to either `CancelledError` (when does it cancel? What is the state of the device?). 

I am getting sidetracked lol. My original point was that we don't want to catch bare Exceptions outside of the config flow. So we'd rather catch the individual types than have a catch-all.